optuna/pruners/_wilcoxon.py

```diff
@@ -0,0 +1,189 @@
+
+            all_steps = list(t.intermediate_values.keys())
+            if all_steps != list(range(1, step + 1)):
+                return None
+
+            intermediate_average_values = np.array(
+                [0] + [t.intermediate_values[step] for step in all_steps]
+            )
+            step_values = np.diff(intermediate_average_values * np.arange(step + 1))
+            return step_values
+
+        step = len(trial.intermediate_values)
+        step_values = extract_step_values(trial)
+        if step_values is None:
+            raise ValueError(
+                "WilcoxonPruner requires intermediate average values to be "
+                "given at step [1, 2, ..., n], but got "
+                f"{list(trial.intermediate_values.keys())} for trial {trial.number})."
+            )
+
+        if np.any(np.isnan(step_values)):
+            warnings.warn(
+                f"The intermediate values of the current trial (trial {trial.number}) "
+                f"contain NaNs. WilcoxonPruner will not prune this trial."
+            )
+            return False
+
+        if step <= self._n_startup_steps:
+            return False
+
+        try:
+            best_trial = study.best_trial
+        except ValueError:
+            return False
+
+        best_step_values = extract_step_values(best_trial)
+        if best_step_values is None:
+            warnings.warn(
+                "WilcoxonPruner requires intermediate average values to be "
+                "given at steps [1, 2, ..., n], but the best trial "
+                f"(trial {best_trial.number}) has intermediate values at "
+                f"steps {list(best_trial.intermediate_values.keys())}. "
+                "WilcoxonPruner will not prune the current trial."
+            )
+            return False
+
+        if len(best_step_values) < len(step_values):
+            warnings.warn(
+                f"The best trial (trial {best_trial.number}) has less "
+                f"({len(best_step_values)}) steps than the current trial. Only the "
+                f"first {len(best_step_values)} intermediate values will be compared."
+            )
+            step_values = step_values[: len(best_step_values)]
+        elif len(best_step_values) > len(step_values):
+            best_step_values = best_step_values[: len(step_values)]
+
+        if np.any(np.isnan(best_step_values)):
+            warnings.warn(
+                f"The intermediate values of the best trial (trial {best_trial.number}) "
+                f"contain NaNs. WilcoxonPruner will not prune the current trial."
+            )
+            return False
+
+        diffs = step_values - best_step_values
+        diffs[np.isnan(diffs)] = 0  # inf - inf or (-inf) - (-inf)
+        if study.direction == StudyDirection.MAXIMIZE:
+            diffs *= -1
+
+        if len(diffs) == 0:
+            return False
+
+        p = ss.wilcoxon(diffs, alternative="greater", zero_method="zsplit").pvalue
```

Comment:
Can we use the `alternative="less"` option instead of `diff *= -1` for maximization?
https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.wilcoxon.html